papas/aliceproto/aliceblockbuilder.py

```diff
@@ -0,0 +1,206 @@
+
+import itertools
+from PFfloodfill import FloodFill
+from enum import Enum
+#import sys
+#Get the DAGtool
+
+from DAG import Node, BreadthFirstSearchIterative,DAGFloodfill
+from heppy.papas.aliceproto.Identifier import Identifier
+from heppy.papas.pfalgo.distance import Distance
+ 
+class Event(object):
+    def __init__(self):
+        self.simParticles=dict()
+        self.recParticles=dict()
+        self.ECALclusters=dict()
+        self.HCALclusters=dict()
+        self.tracks=dict()           #tracks to be used in reconstruction
+        self.historyNodes=dict()  #Nodes used in simulation/reconstruction (contain uniqueid)
+        #self.recHistoryNodes=dict() #Nodes used in reconstruction (contain uniqueid)
+        self.edgeNodes=dict()     #Contains Edge distances between nodes
+        
+        #dictionary with uniqueids
+        self.blocks=dict()           #Blocks made in reconstuction
+        #self.edgedata=dict() 
+    
+
+def MergeNodes (Nodes1 , Nodes2) :
+    
+    for label, node in Nodes2.iteritems() :
+        found=False
+        if Nodes1.find(label) :
+            Nodes1[label].UpdateNode(node)
+
+          
+
+    
+
+
+class BlockBuilder(object):
```

Comment:
generally speaking, we should have explicit dostrings for all classes, and all methods.